Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandItemsStackSplit.java

```diff
@@ -23,61 +28,93 @@ public CommandItemsStackSplit(ODPDBAccess db, HttpServletRequest request, HttpSe
-			
-			if (GameUtils.equals(splitItem.getProperty("containerKey"), character.getKey()) == false) {
-				throw new UserErrorMessage("Item does not belong to character.");
-			}
-			if (!splitItem.hasProperty("quantity")) {
-				throw new UserErrorMessage("You can only split stackable items.");
-			}
-			splitItemQuantity = (Long) splitItem.getProperty("quantity");
-			if (splitItemQuantity==null){
-				throw new UserErrorMessage("You can only split stackable items.");
-			}
-			if (splitItemQuantity < 2) {
-				throw new UserErrorMessage("You can only split stacks of two or more items.");
-			}
 			
 			Long stackSize;
+			Long stacks;
 			try {
 				stackSize = tryParseId(parameters, "stackSize");
-			} catch (NumberFormatException e) {
+				stacks = tryParseId(parameters, "stacks");
+			}
+			catch (NumberFormatException e) {
 				throw new UserErrorMessage("Please input an integer value.");
 			}
-			if (stackSize <= 0) {
+			
+			if (stackSize <= 0 || stacks <= 0) 
 				throw new UserErrorMessage("Please input a positive integer value.");
+			
+			if(batchItems.size() > 1 && stacks > 1) 
+				throw new UserErrorMessage("You can't split multiple items into multiple stacks.");
+			
+			if(stacks > 24) 
+				throw new UserErrorMessage("Too many stacks. Max is 24.");
+			
+			//iterate through all of our items.
+			for(CachedEntity splitItem : batchItems) {	
+				if (CommonChecks.isItemCustom(splitItem)) {
+					sendError("You cannot split a custom item.", splitItem);
+					continue;
+				}
+				if (GameUtils.equals(splitItem.getProperty("containerKey"), character.getKey()) == false) {
+					sendError("Item does not belong to character.", splitItem);
+					continue;
+				}
+				if (!splitItem.hasProperty("quantity")) {
+					sendError("You can only split stackable items.", splitItem);
+					continue;
+				}
+				splitItemQuantity = (Long) splitItem.getProperty("quantity");
+				if (splitItemQuantity==null) {
+					sendError("You can only split stackable items.", splitItem);
+					continue;
+				}
+				if (splitItemQuantity < 2) {
+					sendError("You can only split stacks of two or more items.", splitItem);
+					continue;
+				}
+				if(stackSize*stacks >= splitItemQuantity) {
+					sendError("Given " + stacks + " stacks and a stack size of " + stackSize + ", you don't have enough of this item.", splitItem);
+					continue;
+				}
+				
+				long newStackSize = splitItemQuantity - (stackSize*stacks);
+				ds.preallocateIdsFor(splitItem.getKind(), stacks.intValue());
+				
+				//Create all of the new item stacks.
+				for(int i = 0; i < stacks.intValue(); i++) {
+					CachedEntity newStack = new CachedEntity(splitItem.getKind(), ds.getPreallocatedIdFor(splitItem.getKind()));
+					CachedDatastoreService.copyFieldValues(splitItem, newStack);
+					
+					newStack.setProperty("quantity", stackSize);
+					ds.put(newStack);
+				}
+				
+				splitItem.setProperty("quantity", newStackSize);
+				ds.put(splitItem);
+				
```

Comment:
What happens to the original item if it evenly divides into the stacks. 100 item, split into 10 stacks of 10. You'd end up with 11 items, one of which has 0 quantity.

Reply:
Nevermind. Looks like you're preventing new stack size from being 0 by requiring that the stack can be created.